src/main/java/database/Database.java

```diff
@@ -0,0 +1,49 @@
+package main.java.database;
+
+import main.java.model.Notification;
+import main.java.model.User;
+
+public class Database {
+    private static Database instance;
+    private User currentUser;
+    private static final String usersFolderPath = "src/main/resources/users/";
+    private static final String reportsFolderPath = "src/main/resources/reports/";
+    private static final String notificationFolderPath = "src/main/resources/notifications/";
+    private final ReportsDatabase reportsDatabase;
+    private final UsersDatabase usersDatabase;
```

Comment:
I would suggest replacing separate classes for Reports and Users with `FileDatabase<Report>` and `FileDatabase<User>` respectively. This way, we can still keep the requirement of having a nested class in FileDatabase (FileManager), without duplicating essentially the same code differing only in the type of entity to save.